addons/purchase/purchase_view.xml

```diff
@@ -565,6 +555,7 @@
                     <field name="order_id"/>
                     <field name="product_id"/>
                     <field name="partner_id" string="Vendor" filter_domain="[('partner_id', 'child_of', self)]"/>
+                    <filter string="Confirmed" domain="[('state', 'in', ['confirmed', 'done'])]" name="confirmed"/>
```

Comment:
wrong filter key. in Purchase for confirm state we have ''approved' key. you have to check state for 'approved' key like
domain="[('state', 'in', ['approved', 'done'])]"   ;)